File: admin-ui/src/app/dto/victorina.ts
import {Question} from "./question";
import {Channel} from "./channel";

export class Victorina {

  constructor(public id: unknown,
              public name: string,
              public title: string,
              public questions: Question[],
              public ownerId: string,
              public channel: Channel | null,
              public winnerId: string | null,
              public rightsAnsweredUserId: Array<String>|null,
              public isManyAnswer:boolean,
              public numberParticipants:number,
              public isActive:boolean,
              public hasPrize:Boolean,
              public startDate: Date | null,
              public endDate: Date | null) {

  }

  toString():string{
    return `Victorina[name=${this.name}; title=${this.title}; Cannel=${this.questions};
    startDate=${this.startDate}; endDate=${this.endDate}]`
  }

}

File: bot/src/main/kotlin/com/nimko/bot/models/Victorina.kt
package com.nimko.bot.models

import java.time.LocalDateTime


data class Victorina(
    val id:String?,
    val name:String,
    val title:String,
    val questions:Array<Question>,
    val ownerId:String,
    val channel:Channel?,
    val isManyAnswer:Boolean,
    var isActive:Boolean,
    val startDate:LocalDateTime,
    val endDate:LocalDateTime
) {
}